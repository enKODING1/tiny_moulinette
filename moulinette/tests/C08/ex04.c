#include <stdio.h>
#include <stdlib.h>
#include "ft_stock_str.h"

struct s_stock_str *ft_strs_to_tab(int ac, char **av);

void test(t_stock_str *par)
{
    printf("---\n");
    if (par == NULL)
    {
        printf("NULL\n");
        return;
    }
    for (int i = 0; par[i].str != 0; i++)
    {
        printf("%s\n", par[i].str);
        printf("%d\n", par[i].size);
        printf("%s\n", par[i].copy);
    }
}

int main(void)
{
    char **arr = (char **)malloc(6 * sizeof(char *));

    arr[0] = "Hello";
    arr[1] = "42";
    arr[2] = "Gyeongsan";
    arr[3] = "Fucking";
    arr[4] = "So";
    arr[5] = "Hot";

    test(ft_strs_to_tab(-1, NULL));
    test(ft_strs_to_tab(0, NULL));
    test(ft_strs_to_tab(1, NULL));
    test(ft_strs_to_tab(-1, arr));
    test(ft_strs_to_tab(0, arr));
    test(ft_strs_to_tab(1, arr));
    test(ft_strs_to_tab(2, arr));
    test(ft_strs_to_tab(3, arr));
    test(ft_strs_to_tab(4, arr));
    test(ft_strs_to_tab(5, arr));
    test(ft_strs_to_tab(6, arr));
    return 0;
}
