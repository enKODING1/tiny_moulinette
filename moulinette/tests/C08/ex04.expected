---
NULL
---
NULL
---
NULL
---
NULL
---
---
Hello
5
Hello
---
Hello
5
Hello
42
2
42
---
Hello
5
Hello
42
2
42
Gyeongsan
9
Gyeongsan
---
Hello
5
Hello
42
2
42
Gyeongsan
9
Gyeongsan
So
2
So
---
Hello
5
Hello
42
2
42
Gyeongsan
9
Gyeongsan
So
2
So
Hot
3
Hot
---
Hello
5
Hello
42
2
42
Gyeongsan
9
Gyeongsan
So
2
So
Hot
3
Hot
Place
5
Place
